jaeger_client/sampler.py

```diff
@@ -262,6 +290,29 @@ def is_sampled(self, trace_id, operation=''):
             return sampler.is_sampled(trace_id, operation)
         return sampler.is_sampled(trace_id, operation)
 
+    def update(self, strategies):
+        # (NB) This function should only be called while holding a Write lock.
```

Comment:
look @ line 430, it's kinda weird to pass the lock to and fro. For new logic will be:

`if instance of AdaptiveSampler, then pass the lock inside; else lock and update sampler.`

Rather than just lock in one place.